qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -318,28 +341,209 @@ def _mapper(self, circuit_graph, coupling_graph, trials=20):
-                layout = best_layout
-
-                # Update the DAG
-                if not self.fake_run:
-                    self._layer_update(
-                        dagcircuit_output, layerlist[i], best_layout, best_depth, best_circuit
-                    )
-
         # This is the final edgemap. We might use it to correctly replace
         # any measurements that needed to be removed earlier.
-        logger.debug("mapper: self.trivial_layout = %s", self.trivial_layout)
+        logger.debug("mapper: self.initial_layout = %s", self.initial_layout)
         logger.debug("mapper: layout = %s", layout)
 
+        self.property_set["final_layout"] = layout
         if self.fake_run:
-            self.property_set["final_layout"] = layout
             return circuit_graph
         return dagcircuit_output
+
+    def _controlflow_layer_update(self, dagcircuit_output, layer_dag, current_layout, root_dag):
+        """
+        Updates the new dagcircuit with a routed control flow operation.
+
+        Args:
+           dagcircuit_output (DAGCircuit): dagcircuit that is being built with routed operations.
+           layer_dag (DAGCircuit): layer to route containing a single controlflow operation.
+           current_layout (Layout): current layout coming into this layer.
+           root_dag (DAGCircuit): root dag of pass
+
+        Returns:
+           Layout: updated layout after this layer has been routed.
+
+        Raises:
+            TranspilerError: if layer_dag does not contain a recognized ControlFlowOp.
+
+        """
+        cf_opnode = layer_dag.op_nodes()[0]
+        if isinstance(cf_opnode.op, IfElseOp):
+            updated_ctrl_op, cf_layout, idle_qubits = self._route_control_flow_multiblock(
+                cf_opnode, current_layout, root_dag
+            )
+        elif isinstance(cf_opnode.op, (ForLoopOp, WhileLoopOp)):
+            updated_ctrl_op, cf_layout, idle_qubits = self._route_control_flow_looping(
+                cf_opnode, current_layout, root_dag
+            )
+        else:
+            raise TranspilerError(f"unsupported control flow operation: {cf_opnode}")
+        if self.fake_run:
+            return cf_layout
+
+        cf_layer_dag = DAGCircuit()
+        cf_qubits = [qubit for qubit in root_dag.qubits if qubit not in idle_qubits]
+        qreg = QuantumRegister(len(cf_qubits), "q")
+        cf_layer_dag.add_qreg(qreg)
+        for creg in layer_dag.cregs.values():
+            cf_layer_dag.add_creg(creg)
+        cf_layer_dag.apply_operation_back(updated_ctrl_op, cf_layer_dag.qubits, cf_opnode.cargs)
+        target_qubits = [qubit for qubit in dagcircuit_output.qubits if qubit not in idle_qubits]
+        order = current_layout.reorder_bits(target_qubits)
+        dagcircuit_output.compose(cf_layer_dag, qubits=order)
+        return cf_layout
+
+    def _new_seed(self):
+        """Get a seed for a new RNG instance."""
+        return self.rng.integers(0x7FFF_FFFF_FFFF_FFFF)
```

Comment:
I guess there's not constant we could use for signed 32bit int max. Maybe a comment to explain it would be nice, but it was obvious to me at least.

Reply:
oh yeah, I didn't really think. There's `np.iinfo(np.int64).max`, I suppose?